sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/BeamZetaSqlCatalog.java

```diff
@@ -156,6 +159,48 @@ void addFunction(ResolvedNodes.ResolvedCreateFunctionStmt createFunctionStmt) {
             ImmutableList.of(createFunctionStmt.getSignature())));
   }
 
+  void validateJavaUdf(ResolvedNodes.ResolvedCreateFunctionStmt createFunctionStmt) {
+    for (FunctionArgumentType argumentType :
+        createFunctionStmt.getSignature().getFunctionArgumentList()) {
+      Type type = argumentType.getType();
+      if (type == null) {
+        throw new UnsupportedOperationException("UDF templated argument types are not supported.");
+      }
+      validateJavaUdfZetaSqlType(type);
+    }
+    if (createFunctionStmt.getReturnType() == null) {
+      throw new NullPointerException("UDF return type must not be null.");
```

Comment:
This is going to count as a crash in our SLO monitoring. Can you actually get here or is this rejected by the parser? If you can actually get here you might throw a `UnsupportedOperationException`, if you can't actually get here consider `IllegalArgumentException` (which will still count as a crash but makes it clear it isn't a problem with this code).

Reply:
I don't think you can actually get here. The parser rejects non-SQL UDFs that don't specify a return type.

I changed it to IllegalArgumentException.

Reply:
I also made a PR to add the test I wrote to verify this: #14484